### PR TITLE
Gruntwork patcher

### PR DESCRIPTION
**Pull request by lubiarz-oliwia**

### CODE REVIEW

**Comment by klijakub** [comment]
@lubiarz-oliwia 
Export this image in higher quality 
image
image

File: assets/css/global.scss [hunk 1]
```diff
@@ -1299,6 +1348,26 @@ $padding-section-vertical-desktop: 60px;
...
+}
+.floating-img-patcher {
+  position: absolute;
+  right: 0;
+  @media (max-width: $screen-md-max) {
+    left: 50%;
+    transform: translate(-50%);
+  }
+}
+
```
Comment (klijakub) [inline assets/css/global.scss]: Should be in `patcher.scss`

File: assets/css/pages/patcher.scss [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+      border: 0;
+      border-radius: 500px;
+      width: 100%;
+      max-width: 100%;
+      transition: box-shadow 0.2s ease-in-out;
+
+      // Chrome
+      @media screen and (-webkit-min-device-pixel-ratio: 0) {
+        & {
+          // overflow: hidden;
```
Comment (klijakub) [inline assets/css/pages/patcher.scss]: Remove comment

File: assets/css/pages/patcher.scss [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+          height: 9px;
+          -webkit-appearance: none;
+          background-color: transparent;
+          border: #04bedf solid 1px;
+        }
+
+        &::-webkit-slider-runnable-track {
+          height: 10px;
+          -webkit-appearance: none;
+          // margin-top: -1px;
```
Comment (klijakub) [inline assets/css/pages/patcher.scss]: Remove comment

File: assets/css/pages/patcher.scss [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+            #07a5fe 86.8%,
+            #06a2ff 86.8%,
+            #07a5fe 86.8%
+          );
+          border-radius: 13px;
+        }
+
+        &:active::-webkit-slider-thumb {
+          //background: #1E2530;
+        }
```
Comment (klijakub) [inline assets/css/pages/patcher.scss]: Do we need this?

**Review by klijakub — CHANGES_REQUESTED** [review]

**Comment by klijakub** [comment]
image
Icon isn't sharp, poor quality export

**Comment by klijakub** [comment]
image
Fix hover issue

**Comment by klijakub** [comment]
Mobile view issue:
image

File: assets/js/asciinema-player.min.js [hunk 5]
```diff
@@ -0,0 +1 @@
...7)}:{driver:"asciicast",url:A}),void 0===A.driver&&(A.driver="asciicast");var g=new Map([["asciicast",Rg],["websocket",bg],["eventsource",Hg],["test",Jg]]);if("function"==typeof A)return A;if(g.has(A.driver)){var I=g.get(A.driver);return function(g,B){return I(A,g,B)}}throw"unsupported driver: ".concat(JSON.stringify(A))}return A.create=function(A,g){var I,B,Q=arguments.length>2&&void 0!==arguments[2]?arguments[2]:{},C=new WA(mg(A),{cols:Q.cols,rows:Q.rows,loop:Q.loop,speed:Q.speed,preload:Q.preload,startAt:Q.startAt,poster:Q.poster,idleTimeLimit:Q.idleTimeLimit}),E={core:C,cols:Q.cols,rows:Q.rows,fit:Q.fit,autoPlay:null!==(I=Q.autoPlay)&&void 0!==I?I:Q.autoplay,terminalFontSize:Q.terminalFontSize,terminalFontFamily:Q.terminalFontFamily,terminalLineHeight:Q.terminalLineHeight,theme:Q.theme},t=gA((function(){return B=X(dg,E)}),g),e={el:B,dispose:t,getCurrentTime:function(){return C.getCurrentTime()},getDuration:function(){return C.getDuration()},play:function(){return C.play()},pause:function(){return C.pause()},seek:function(A){return C.seek(A)},addEventListener:function(A,g){return C.addEventListener(A,g.bind(e))}};return e},Object.defineProperty(A,"__esModule",{value:!0}),A}({});
```
Comment (klijakub) [inline assets/js/asciinema-player.min.js]: Remove console.debug

**Comment by klijakub** [comment]
Export in better quality:
image

File: _includes/scripts.html [hunk 6]
```diff
@@ -8,6 +8,55 @@
...
+<script>
+  AsciinemaPlayer.create(
+    "{{ site.assets_base_url }}asciinema/Patcher.cast",
+    document.getElementById("patcher-player-cast-1"),
+    { cols: 100, rows: 20, autoPlay: true }
+  );
+  AsciinemaPlayer.create(
+    "{{ site.assets_base_url }}asciinema/Patcher.cast",
+    document.getElementById("patcher-player-cast-2"),
+    { cols: 101, rows: 24, autoPlay: true }
```
Comment (ArkuVonSymfon) [inline _includes/scripts.html]: Was this one done on purpose with different cols and rows? Any others have the same like "151" and "36"

File: _includes/styles.html [hunk 7]
```diff
@@ -1 +1,6 @@
 <link rel="stylesheet" href="{{ site.assets_base_url }}css/app.css" />
+<link
+  rel="stylesheet"
+  type="text/css"
+  href="{{ site.assets_base_url }}css/asciinema-player.css"
+/>
```
Comment (ArkuVonSymfon) [inline _includes/styles.html]: end line

File: _data/patcher.yml [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+  planned: GitLab, BitBucket
+- name: Operating system
+  beta: macOS, Linux
+  planned: Windows
+- name: Update method
+  beta: Update the folders you want with the CLI
+  planned: Set up automatic updates with the web UI
+- name: Availability
+  beta: Part of the Gruntwork Subscription
+  planned: Standalone product
```
Comment (ArkuVonSymfon) [inline _data/patcher.yml]: end line

File: assets/js/slider.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+      left: (rangePercent * 100) / numberOfModulesIn2021 + "%",
+    });
+    $(".progress-bar").css({
+      width: (rangePercent * 100) / numberOfModulesIn2021 + "%",
+    });
+    $(".updates").html(calculatedUpdates);
+    $(".hours").html(calculatedHours);
+    $(".money-saved").html("$" + calculatedSavedMoney);
+  });
+});
```
Comment (ArkuVonSymfon) [inline assets/js/slider.js]: end line

File: pages/patcher/_hero.html [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+          ga-event-action="Click Contact Us(Landing zone for aws Page)"
+          ga-event-label="User clicked contact button in Landing zone for aws page">{{ link.title }}Get Started</a>
+      </p>
+    </div>
+    <div class="col-lg-7">
+      <img onclick="toggleVideo()" class="floating-img-patcher" src="/assets/img/patcher/patcher-hero@3x.png"
+        style="z-index: 999; cursor: pointer; max-height: 505px;" />
+    </div>
+  </div>
+</div>
```
Comment (ArkuVonSymfon) [inline pages/patcher/_hero.html]: end line

File: pages/patcher/_how-it-works.html [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+            document.querySelector('.section-work-without-patcher').classList.toggle('section-work-active')
+            document.querySelector('.without-patcher').classList.remove('active-btn')
+        } else {
+            document.querySelector('.section-work-with-patcher').classList.remove('section-work-active')
+            document.querySelector('.with-patcher').classList.remove('active-btn')
+            document.querySelector('.section-work-without-patcher').classList.add('section-work-active')
+            document.querySelector('.without-patcher').classList.add('active-btn')
+        }
+    }
+</script>
```
Comment (ArkuVonSymfon) [inline pages/patcher/_how-it-works.html]: end line

File: pages/patcher/_patcher-table.html [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+                {{ package.planned }}
+              </td>
+            </tr>
+            {% endfor %}
+          </tbody>
+        </table>
+      </div>
+    </div>
+  </div>
+</div>
```
Comment (ArkuVonSymfon) [inline pages/patcher/_patcher-table.html]: end line

File: pages/patcher/_video.html [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+    </video>
+</div>
+
+<script>
+    function toggleVideo() {
+        document
+            .querySelector(".video-wrapper")
+            .classList.toggle("video-wrapper-active");
+    }
+</script>
```
Comment (ArkuVonSymfon) [inline pages/patcher/_video.html]: end line

File: pages/patcher/_slider.html [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+    </div>
+  </div>
+  <div class="col-md-4">
+    <p class="additional-info">
+      *Estimate of developers' average earnings based on
+      <a href="ttps://money.usnews.com/careers/best-jobs/software-developer/salary" target="_blank">U.S. News
+        analysis.</a>
+    </p>
+  </div>
+</div>
```
Comment (ArkuVonSymfon) [inline pages/patcher/_slider.html]: end line

**Review by ArkuVonSymfon — CHANGES_REQUESTED** [review]

**Pull request closed without merging**